### PR TITLE
Clippy tests

### PR DESCRIPTION
**Pull request by martinthomson**
VS Code runs clippy on tests.  I don't know how to do that from the command line (or else I would change our CI setup as well), but these are the problems I found and fixed.

The only thing I didn't fix were a couple of functions that were past the cognitive complexity limit.

### CODE REVIEW

File: neqo-http3/src/connection.rs [hunk 1]
```diff
@@ -1047,22 +1047,25 @@ mod tests {
...
         // connection on the other side so that we can check what the http3
         // side sends and also to simulate an incorrectly behaving http3
         // server/client.
 
-        let mut hconn;
-        let mut neqo_trans_conn;
-        if client {
-            hconn = Http3Connection::new(default_client(), 100, 100, None);
-            neqo_trans_conn = default_server();
+        let (mut hconn, mut neqo_trans_conn) = if client {
```
Comment (martinthomson) [inline neqo-http3/src/connection.rs]: I don't like how clippy warns about this one.  But I'm not in a mood to suppress the change, even if it isn't great.

File: neqo-http3/src/connection.rs [hunk 2]
```diff
@@ -1332,16 +1333,14 @@ mod tests {
         let events = neqo_trans_conn.events();
         let mut stop_sending_event_found = false;
         for e in events {
```
Comment (martinthomson) [inline neqo-http3/src/connection.rs]: This is a common pattern.  I was uncertain about whether to change this to use Iterator::any, which would be better for looking for the event.  But that would be lots of typing.

**Comment by o0Ignition0o** [comment]
> I don't know how to do that from the command line (or else I would change our CI setup as well)

[it seems](https://github.com/rust-lang/rust-clippy/issues/1436#issuecomment-462059561) `cargo clippy --tests` does the trick :)

**Comment by martinthomson** [comment]
> [it seems](https://github.com/rust-lang/rust-clippy/issues/1436#issuecomment-462059561) `cargo clippy --tests` does the trick :)

I went with something a little broader than that. `cargo clippy --all-features --all-targets` gets more than just the one set of configs.  Gee it would be nice if `--help` actually listed these things.

**Comment by martinthomson** [comment]
> Gee it would be nice if `--help` actually listed these things.

Oh, it does:

```
Other options are the same as `cargo check`.
```

**Pull request merged**